charts/cluster-api-runtime-extensions-nutanix/defaultclusterclasses/nutanix-cluster-class.yaml

```diff
@@ -0,0 +1,305 @@
+spec:
+  controlPlane:
+    machineHealthCheck:
+      maxUnhealthy: 40%
+      nodeStartupTimeout: 10m
+      unhealthyConditions:
+      - status: "False"
+        timeout: 300s
+        type: Ready
+      - status: Unknown
+        timeout: 300s
+        type: Ready
+      - status: "True"
+        timeout: 300s
+        type: MemoryPressure
+      - status: "True"
+        timeout: 300s
+        type: DiskPressure
+      - status: "True"
+        timeout: 300s
+        type: PIDPressure
+      - status: "True"
+        timeout: 300s
+        type: NetworkUnavailable
+    machineInfrastructure:
+      ref:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: NutanixMachineTemplate
+        name: nutanix-quick-start-cp-nmt
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: KubeadmControlPlaneTemplate
+      name: nutanix-quick-start-kcpt
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: NutanixClusterTemplate
+      name: nutanix-quick-start-nct
+  patches:
+  - external:
+      discoverVariablesExtension: nutanixclusterconfigvars.cluster-api-runtime-extensions-nutanix
+      generateExtension: nutanixclusterconfigpatch.cluster-api-runtime-extensions-nutanix
+    name: cluster-config
+  - external:
+      discoverVariablesExtension: nutanixworkerconfigvars.cluster-api-runtime-extensions-nutanix
+      generateExtension: nutanixworkerconfigpatch.cluster-api-runtime-extensions-nutanix
+    name: worker-config
+  workers:
+    machineDeployments:
+    - class: nutanix-quick-start-worker
+      machineHealthCheck:
+        maxUnhealthy: 40%
+        nodeStartupTimeout: 10m
+        unhealthyConditions:
+        - status: "False"
+          timeout: 300s
+          type: Ready
+        - status: Unknown
+          timeout: 300s
+          type: Ready
+        - status: "True"
+          timeout: 300s
+          type: MemoryPressure
+        - status: "True"
+          timeout: 300s
+          type: DiskPressure
+        - status: "True"
+          timeout: 300s
+          type: PIDPressure
+        - status: "True"
+          timeout: 300s
+          type: NetworkUnavailable
+      template:
+        bootstrap:
+          ref:
+            apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+            kind: KubeadmConfigTemplate
+            name: nutanix-quick-start-kcfg-0
+        infrastructure:
+          ref:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+            kind: NutanixMachineTemplate
+            name: nutanix-quick-start-md-nmt
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlaneTemplate
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: nutanix
+  name: nutanix-quick-start-kcpt
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        clusterConfiguration:
+          apiServer:
+            certSANs:
+            - localhost
+            - 127.0.0.1
+            - 0.0.0.0
```

Comment:
This should not be in the base template.

Reply:
https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/blob/main/docs/content/customization/generic/extra-apiserver-cert-sans.md

this is how we can set it through the cluster variable, and our default Cluster implementation can have this. Or we can add this in automatically via code.

Reply:
we already have PR open https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/372 for capx.

Reply:
This shouldn't be a json patch and should instead be a done with GeneratePatches

Reply:
let's keep these open for more discussion.

Reply:
Just one more point, this clusterclass isn't meant to be used by CAPX. It's an opinionated clusterclass for CAREN. We don't need to worry about the other consumers here.

Reply:
I think plan is clear:
- add variable in capx https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/372
- in subsequent PR use generic https://d2iq-labs.github.io/cluster-api-runtime-extensions-nutanix/customization/generic/extra-apiserver-cert-sans/ to update certSAN 

Is there anything open in your mind?